### ISSUE TITLE
VMAlert reporting problem with remote write to victoria metrics

### ISSUE THREAD

**Issue by cskinfill** [VictoriaMetrics/VictoriaMetrics#660]
**Describe the bug**
In the vmalert logs I frequently see logs like this, its not clear to me if this is a bug, if my victoria metrics server is overwhelmed (doesn't seem like it is ... but i'm still figuring out how this behaves), or some other problem with vmalert talking to vm server.

> 2020-07-30T13:25:30.920Z	error	VictoriaMetrics/app/vmalert/remotewrite/remotewrite.go:197	error getting response from http://victoriametrics-victoria-metrics-single-server:8428/api/v1/write:Post "http://victoriametrics-victoria-metrics-single-server:8428/api/v1/write": EOF

I'm running vm server in k8s and vmalert in the same cluster and namespace.  Both installed by Helm.

I noticed this in the context of debugging why vmalert isn't sending alerts to alertmanager.

**To Reproduce**
Shows frequently in the vmalert logs

**Expected behavior**
If this is a bug, what's the cause?  Its not clear.

**Version**
vmagent is running container `victoriametrics/vmalert:v1.37.4`
vmserver is running container `victoriametrics/victoria-metrics:v1.38.1`

**Used command-line flags**
Command-line flags are listed as `flag{name="httpListenAddr", value=":443"} 1` lines at `/metrics` page.
See the following docs for details:
for vmserver pod
```
      --retentionPeriod=3
      --storageDataPath=/storage
```
for vmalert pod
```
      -rule=/config/alert-rules.yaml
      -datasource.url=http://victoriametrics-victoria-metrics-single-server:8428/
      -notifier.url=http://vmalert-alertmanager:9093
      -remoteRead.url=http://victoriametrics-victoria-metrics-single-server:8428/
      -remoteWrite.url=http://victoriametrics-victoria-metrics-single-server:8428/
```

Is this a problem with vmalert?  or is vmserver overwhelmed?
Thanks (btw, really liking victoria metrics)

**Comment by valyala** [comment]
This error may be already fixed in the commit 703def4b2e9b82ebb6aa6aaa640730b0781d7110 , which has been included in `vmalert` starting from [v1.38.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Try upgrading `vmalert` to the latest available version from the [releases page](https://github.com/VictoriaMetrics/VictoriaMetrics/releases) and verifying whether the issue is fixed there.

**Comment by hagen1778** [comment]
Hi @cskinfill! Like @valyala pointed `vmalert` wasn't tolerant to network issues in previous versions. Please update to latest version and see if situation improves. Btw, in the latest version you can track additional `vmalert` metrics for remote write component:
* vmalert_remotewrite_sent_rows_total
* vmalert_remotewrite_sent_bytes_total
* vmalert_remotewrite_dropped_rows_total
* vmalert_remotewrite_dropped_bytes_total

**Comment by valyala** [comment]
FYI, the latest version for `vmalert` is [v1.39.2](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Comment by valyala** [comment]
Closing this issue as fixed (I assume it has been fixed since v1.38.0 with the commit https://github.com/VictoriaMetrics/VictoriaMetrics/commit/703def4b2e9b82ebb6aa6aaa640730b0781d7110 ). @cskinfill , feel free re-opening the issue if you see it in the latest versions of `vmalert` - see https://github.com/VictoriaMetrics/VictoriaMetrics/releases
